src/components/VideosList/VideosList.component.jsx

```diff
@@ -34,25 +33,29 @@ const MarginX = styled.div`
   margin: 0 0.75rem;
 `;
 
-const VideosList = () =>
-  items.map(({ id, snippet }) => (
-    <VideoCard key={id.videoId}>
-      <figure>
-        <img src={snippet.thumbnails.medium.url} alt={`${snippet.title} thumbnail`} />
-      </figure>
-      <Wrapper>
-        <ChannelAvatar src={channel.thumbnails.default.url} alt={channel.channelTitle} />
-        <MarginX className="margin-x">
-          <Link href={`https://www.youtube.com/watch?v=${id.videoId}`}>
-            {snippet.title}
-          </Link>
-          <div>
-            <Text>{snippet.channelTitle}</Text>
-            <Text>{snippet.publishedAt}</Text>
-          </div>
-        </MarginX>
-      </Wrapper>
-    </VideoCard>
-  ));
+const Video = ({ id, videoData }) => (
+  <VideoCard>
+    <figure>
+      <img src={videoData.thumbnails.medium.url} alt={`${videoData.title} thumbnail`} />
+    </figure>
+    <Wrapper>
+      <ChannelAvatar src={defaultChannelAvatar} alt={videoData.channelTitle} />
+      <MarginX className="margin-x">
+        <StyledLink to={`/${id}`}>{videoData.title}</StyledLink>
+        <div>
+          <Text>{videoData.channelTitle}</Text>
+          <Text>{videoData.publishedAt}</Text>
+        </div>
+      </MarginX>
+    </Wrapper>
+  </VideoCard>
+);
+
+const VideosList = ({ videos }) =>
+  videos
+    .filter(({ snippet }) => snippet !== undefined)
+    .map(({ id, snippet }) => (
+      <Video key={id.videoId} id={id.videoId} videoData={snippet} />
+    ));
```

Comment:
This works well, however it is not a component itself, but an array of Video components. In these cases a good practice is to have instead just the Video component (or VideoThumbnail, VideoInfo, etc.) and do the iteration directly in the parent component.
In this case you would have the `videos.filter(...).map(...)` directly in the Home page and VideoDetail page

It's not a big deal and for this project this works well, but on bigger projects you may want to have custom containers for each list or pass custom props to the thumbnail itself

Reply:
Thanks for the feedback. I've changed Home & VideoDetail component to add the VideoThumbnail compoennt instaed VideosList component.